src/Scenes/GameScene/Views/Editor/Plotter/FloorPlotter.ts

```diff
@@ -0,0 +1,47 @@
+import * as PIXI from 'pixi.js';
+import {PlotterDirectionRestriction, Plotter} from './Plotter';
+import {FloorType} from '../../../../../GameLogic/Enums/FloorType';
+import {TextureStore} from 'evidently-pixi';
+import {EditorConfig} from '../EditorConfig';
+import {GfxConstants} from '../../../../../Core/Constants/GfxConstants';
+import {TurnState} from '../../../../../GameLogic/TurnState';
+import {TurnEventType} from '../../../../../GameLogic/Enums/TurnEventType';
+
+export class FloorPlotter implements Plotter {
+	public floorType: FloorType;
+
+	constructor(floor: FloorType) {
+		this.floorType = floor;
+	}
+
+	public plot(turnState: TurnState, editorConfig: EditorConfig, x: number, y: number): void {
+		const {level} = turnState;
+
+		level.tilesFloor.set(x, y, this.floorType);
+		turnState.addEvent(TurnEventType.TileChanged, [x, y]);
+	}
+
+	public unplot(turnState: TurnState, editorConfig: EditorConfig, x: number, y: number): void {
+		const {level} = turnState;
+
+		level.tilesFloor.set(x, y, FloorType.FloorTile);
+		turnState.addEvent(TurnEventType.TileChanged, [x, y]);
+	}
+
+	public getTexture(editorConfig: EditorConfig, textureStore: TextureStore): PIXI.Texture {
```

Comment:
Should we also add the exit tile to the editor at this point?

Reply:
Good point!